Kudu.Core/Scripts/selectNodeVersion.js

```diff
@@ -34,13 +34,26 @@ function resolveNpmPath(npmRootPath, npmVersion) {
 
 
 function getDefaultNpmVersion(nodeVersionPath) {
-    var npmLinkPath = path.resolve(nodeVersionPath, 'npm.txt');
-    // Determine if there's a link to npm at the node path
-    if (!existsSync(npmLinkPath)) {
-        return;
+    var appSettingNpmVersion = process.env.WEBSITE_NPM_DEFAULT_VERSION;
+
+    // extract the current node version from the path
+    var nodePathSplited = nodeVersionPath.split(path.sep).filter(function(e) { return e; });
+    var currentNodeVersion = nodePathSplited[nodePathSplited.length - 1];
+
+    if (appSettingNpmVersion) {
+        return appSettingNpmVersion;
+    } else if (currentNodeVersion === '4.1.2') {
```

Comment:
we also added v4.2.1 and v5.0 and both map to NPM 3.3.9. Do we need to do similar special handling for them as well?